lib/Doctrine/DBAL/Driver/PDOStatement.php

```diff
@@ -145,9 +200,51 @@ public function fetchAll($fetchMode = null, $fetchArgument = null, $ctorArgs = n
     public function fetchColumn($columnIndex = 0)
     {
         try {
-            return parent::fetchColumn($columnIndex);
+            return $this->stmt->fetchColumn($columnIndex);
         } catch (\PDOException $exception) {
             throw new PDOException($exception);
         }
     }
+
+    /**
+     * Converts DBAL parameter type to PDO parameter type
+     *
+     * @param int $type Parameter type
+     * @return int
+     */
+    private function convertParamType(int $type) : int
+    {
+        if ( ! isset(self::$paramTypeMap[$type])) {
+            throw new \InvalidArgumentException('Invalid parameter type: ' . $type);
+        }
+
+        return self::$paramTypeMap[$type];
+    }
+
+    /**
+     * Converts DBAL fetch mode to PDO fetch mode
+     *
+     * @param int|null $fetchMode Fetch mode
+     * @return int|null
+     */
+    private function convertFetchMode(?int $fetchMode) : ?int
+    {
+        if ($fetchMode === null) {
+            return null;
+        }
+
+        if ( ! isset(self::$fetchModeMap[$fetchMode])) {
+            throw new \InvalidArgumentException('Invalid fetch mode: ' . $fetchMode);
+        }
+
+        return self::$fetchModeMap[$fetchMode];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getIterator()
+    {
+        yield from $this->stmt;
```

Comment:
This is interesting, didn't know \PDOStatement is a generator.

Reply:
You can use `yield from` with anything that is `iterable` it doesn't have to be an generator.

Reply:
Oh neat, didn't realize. 👍